### DISCUSSION TITLE
Is it possible to create cli command groups? Or to use hydra with python click?

### DISCUSSION THREAD

**Discussion by vmgustavo** [facebookresearch/hydra#2671] [Q&A]
I found a few issues about this [#1964 #409] but there is not definitive answer to the question. I have a well stablished project structure using click that works fine and I did not want to change much, but I wanted to try hydra with it as my metadata/config file isn't at its best right now. It can be improved and maybe using Hydra would be the way for it.

The two options to achieve that would be:

- replace click with hydra by using cli command groups that hydra would provide (if that exists)
- add hydra to the existing cli command groups programmed with click

Is there any of theese options viable?

**Comment by Jasha10** [comment]
Hydra's `@hydra.main` decorator "owns the command line", meaning that calling a `@main`-decorated function will read `sys.argv`.
I believe that click also "owns the command line".

If you want click to handle the parsing of `sys.argv`, you could use [the Hydra compose API](https://hydra.cc/docs/1.3/advanced/compose_api/) in combination with Click's parsing.
If you want Hydra's `@main` decorator to handle the parsing of `sys.argv`, I don't think there's a straightforward way to integrate with click.

**Comment by vmgustavo** [comment]
Does hydra support cli command groups?

**Reply by Jasha10** [reply]
Not exactly, though you can get some similar functionality by using config groups.
See the "Common Patterns" pages in the docs, e.g. [Configuring Plugins](https://hydra.cc/docs/1.3/patterns/configuring_plugins/).

**Reply by vmgustavo** [reply]
thanks. ill take a look at it

**Comment by vmgustavo** [comment]
I am doing something similar to this: https://github.com/facebookresearch/hydra/issues/1964#issuecomment-1014835636

But I can't figure out a way to make `--hydra-help` work. Is there a function in the API to call the help generator?

**Reply by Jasha10** [reply]
No, I don't think there's a public api for calling --hydra-help when using `compose`.
